### PR TITLE
Final changes to implement azure blob store for online store

### PR DESCRIPTION
**Pull request by Sami1309**
# Description

testing fixes and slight API/bug fixes for azure blob store to run end to end

### Select type(s) of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: runner/materialize.go [hunk 1]
```diff
@@ -15,8 +15,9 @@ import (
 	"github.com/featureform/types"
 )
 
-const MAXIMUM_CHUNK_ROWS int64 = 1024
-const WORKER_IMAGE string = "featureformcom/worker"
+const MAXIMUM_CHUNK_ROWS int64 = 102400
```
Comment (sdreyer) [inline runner/materialize.go:18]: Will this work with 1 million rows? If we have a 10M row dataset this is going to spawn 100 jobs

File: coordinator/coordinator.go [hunk 2]
```diff
@@ -23,6 +23,17 @@ import (
 	"go.etcd.io/etcd/client/v3/concurrency"
 )
 
+func retryWithDelays(name string, retries int, delay time.Duration, idempotentFunction func() error) error {
```
Comment (sdreyer) [inline coordinator/coordinator.go:26]: You might want to use a library for this like https://github.com/avast/retry-go

File: provider/online_test.go [hunk 3]
```diff
@@ -293,6 +294,47 @@ func testEntityNotFound(t *testing.T, store OnlineStore) {
 	}
 }
 
+func testMassTableWrite(t *testing.T, store OnlineStore) {
+	for i := 0; i < 10; i++ {
```
Comment (ahmadnazeri) [inline provider/online_test.go:298]: What's the purpose of this for loop?

File: runner/materialize.go [hunk 4]
```diff
@@ -15,8 +15,9 @@ import (
...
+
+var WORKER_IMAGE string = helpers.GetEnv("WORKER_IMAGE", "featureformcom/worker:0.2.0-rc")
```
Comment (ahmadnazeri) [inline runner/materialize.go:20]: does it ever pull the image from the environment variable? is there a way to pull the latest? I don't think it is a good idea to hardcode version `0.2.0-rc`

File: runner/materialize.go [hunk 5]
```diff
@@ -151,11 +154,13 @@ func (m MaterializeRunner) Run() (types.CompletionWatcher, error) {
 			EnvVars:  envVars,
 			Image:    WORKER_IMAGE,
 			NumTasks: int32(numChunks),
+			Resource: metadata.ResourceID{Name: m.ID.Name, Variant: m.ID.Variant, Type: provider.ProviderToMetadataResourceType[m.ID.Type]},
 		}
 		kubernetesRunner, err := kubernetes.NewKubernetesRunner(kubernetesConfig)
 		if err != nil {
 			return nil, fmt.Errorf("kubernetes runner: %w", err)
 		}
+		fmt.Printf("Kubernetes config env vars: %v\n", envVars)
```
Comment (ahmadnazeri) [inline runner/materialize.go:163]: would this print out the environment variables for the pod that executes the pandas script?

**Review by ahmadnazeri — COMMENTED** [review]
Just a few comments and some clarifications. But generally, looks good to me.

File: kubernetes/kubernetes.go [hunk 6]
```diff
@@ -24,9 +25,12 @@ import (
...
 
-func GetJobName(id metadata.ResourceID) string {
-	return strings.ReplaceAll(fmt.Sprintf("%s-%s-%d", strings.ToLower(id.Name), strings.ToLower(id.Variant), id.Type), "_", ".")
-
+func GetJobName(id metadata.ResourceID, image string) string {
+	jobName := strings.ReplaceAll(fmt.Sprintf("%s-%s-%s-%s-%s", id.Name, id.Variant, id.Type, image, uuid.New().String()), "_", ".")
+	removedSlashes := strings.ReplaceAll(jobName, "/", "")
+	removedColons := strings.ReplaceAll(removedSlashes, ":", "")
+	lowerCase := strings.ToLower(removedColons)
+	return lowerCase[0:int(math.Min(float64(len(lowerCase)), 63))]
```
[the remark this answers is not in the document]
Comment (Sami1309) [reply kubernetes/kubernetes.go:33]: changed

File: kubernetes/kubernetes.go [hunk 7]
```diff
@@ -229,10 +233,12 @@ func NewKubernetesRunner(config KubernetesRunnerConfig) (CronRunner, error) {
 	jobSpec := newJobSpec(config)
 	var jobName string
 	if config.Resource.Name != "" {
-		jobName = GetJobName(config.Resource)
+		jobName = GetJobName(config.Resource, config.Image)
 	} else {
 		cleanUUID := strings.ReplaceAll(uuid.New().String(), "-", "")
-		jobName = fmt.Sprintf("job%s", cleanUUID)
+		jobName = fmt.Sprintf("job__%s", cleanUUID)
+		jobName = jobName[0:int(math.Min(float64(len(jobName)), 63))]
```
[the remark this answers is not in the document]
Comment (Sami1309) [reply kubernetes/kubernetes.go:240]: changed

File: provider/online_test.go (hunk 3 above)
Comment (Sami1309) [reply provider/online_test.go:298]: removed. Was to check if my local environment was just weirdly slow

File: runner/materialize.go (hunk 4 above)
Comment (Sami1309) [reply runner/materialize.go:20]: changed to latest

File: runner/materialize.go (hunk 5 above)
Comment (Sami1309) [reply runner/materialize.go:163]: yes

**Pull request merged**